src/tools/perf/perftest.c

```diff
@@ -361,6 +374,48 @@ static const char *__basename(const char *path)
     return (p == NULL) ? path : p;
 }
 
+static ucs_status_t parse_message_sizes_params(const char *optarg,
+                                               ucx_perf_params_t *params)
+{
+    char *optarg_ptr, *optarg_ptr2;
+    size_t token_num, token_it;
+    const char delim = ',';
+
+    optarg_ptr = (char *)optarg;
+    token_num  = 0;
+    /* count the number of given message sizes */
+    while ((optarg_ptr = strchr(optarg_ptr, delim)) != NULL) {
+        ++optarg_ptr;
+        ++token_num;
+    }
+    ++token_num;
+
+    if (params->msg_size_list) { /* free previously allocated buffer */
+        free(params->msg_size_list);
+    }
+    params->msg_size_list = malloc(sizeof(*params->msg_size_list) * token_num);
+    if (NULL == params->msg_size_list) {
+        return UCS_ERR_NO_MEMORY;
+    }
+
+    optarg_ptr = (char *)optarg;
+    errno = 0;
+    for (token_it = 0; token_it < token_num; ++token_it) {
+        params->msg_size_list[token_it] = strtoul(optarg_ptr, &optarg_ptr2, 10);
```

Comment:
@yosefe  I did this to avoid user confusion in case of `-s 0100,0200` (octal base).
Do we want this behaviour?